pywr-schema/src/timeseries/mod.rs

```diff
@@ -0,0 +1,229 @@
+
+        let array = series.cast(&Float64)?.f64()?.to_ndarray()?.to_owned();
+        let name = format!("{}_{}", name, col);
+
+        match network.get_parameter_index_by_name(&name) {
+            Ok(idx) => Ok(idx),
+            Err(e) => match e {
+                PywrError::ParameterNotFound(_) => {
+                    let p = Array1Parameter::new(&name, array, None);
+                    Ok(network.add_parameter(Box::new(p))?)
+                }
+                _ => Err(TimeseriesError::PywrCore(e)),
+            },
+        }
+    }
+
+    pub fn load_df(
+        &self,
+        network: &mut pywr_core::network::Network,
+        name: &str,
+        domain: &ModelDomain,
+        scenario: &str,
+    ) -> Result<ParameterIndex<f64>, TimeseriesError> {
+        let scenario_group_index = domain
+            .scenarios()
+            .group_index(scenario)
+            .ok_or(TimeseriesError::ScenarioGroupNotFound(scenario.to_string()))?;
+
+        let df = self
+            .timeseries
+            .get(name)
+            .ok_or(TimeseriesError::TimeseriesNotFound(name.to_string()))?;
+
+        let array: Array2<f64> = df.to_ndarray::<Float64Type>(IndexOrder::default()).unwrap();
+        let name = format!("timeseries.{}_{}", name, scenario);
+
+        match network.get_parameter_index_by_name(&name) {
+            Ok(idx) => Ok(idx),
+            Err(e) => match e {
+                PywrError::ParameterNotFound(_) => {
+                    let p = Array2Parameter::new(&name, array, scenario_group_index, None);
+                    Ok(network.add_parameter(Box::new(p))?)
+                }
+                _ => Err(TimeseriesError::PywrCore(e)),
+            },
+        }
+    }
+}
+
+pub fn convert_from_v1_data(df_data: &[TimeseriesV1Data], v1_tables: &Option<TableVec>) -> Vec<Timeseries> {
+    let mut ts = HashMap::new();
+    for data in df_data.iter() {
+        match &data.source {
+            TimeseriesV1Source::Table(name) => {
+                let tables = v1_tables.as_ref().unwrap();
+                let table = tables.iter().find(|t| t.name == *name).unwrap();
+                let name = table.name.clone();
+                if ts.contains_key(&name) {
+                    continue;
+                }
+
+                let time_col = None;
+                let provider = PolarsDataset::new(time_col, table.url.clone());
+
+                ts.insert(
+                    name.clone(),
+                    Timeseries {
+                        meta: ParameterMeta { name, comment: None },
+                        provider: TimeseriesProvider::Polars(provider),
+                    },
+                );
+            }
+            TimeseriesV1Source::Url(url) => {
+                let name = data.name.as_ref().unwrap().clone();
+                if ts.contains_key(&name) {
+                    continue;
+                }
+
+                // TODO: time_col should use the index_col from the v1 data?
```

Comment:
Yes

Reply:
What about this one?

Reply:
oops I started it but didn't finish. Should be done now